tests/units/Services/Table/BuilderTest.php

```diff
@@ -30,235 +30,288 @@ public function setUp(): void
+                ->get('custom')
+                ->get('relation')
+        );
     }
 
     /** @test */
     public function can_get_data_with_flatten()
     {
-        $response = $this->requestResponse([
-            'flatten' => true,
-            "appends" => [
-                "append"
-            ]
-        ]);
+        $this->params['flatten'] = true;
+        $this->params['appends'] = ['custom'];
+
+        $response = $this->requestResponse();
 
-        $this->assertEquals('appended', $response["data"][0]["append.append"]);
+        $this->assertEquals('name', $response->get('data')->first()->get('custom.relation'));
     }
 
     /** @test */
     public function can_get_data_with_enum()
     {
-        $this->testModel->is_active = 1;
-        $this->testModel->save();
+        $this->testModel->update(['is_active' => true]);
 
-        $response = $this->requestResponse([
-            "columns" => [
-                'is_active' => ['name' => 'is_active', 'enum' => BuilderTestEnum::class]
-            ],
-            'meta' => [
-                'enum' => true
-            ]
-        ]);
+        $this->params['columns']['is_active'] = ['name' => 'is_active', 'enum' => BuilderTestEnum::class];
 
-        $this->assertEquals('Active', $response["data"][0]["is_active"]);
+        $this->params['meta']['enum'] = true;
+
+        $response = $this->requestResponse();
+
+        $this->assertEquals(
+            BuilderTestEnum::get($this->testModel->is_active),
+            $response->get('data')->first()->get('is_active')
+        );
     }
 
     /** @test */
     public function can_get_data_with_date()
     {
-        $response = $this->requestResponse([
-            "columns" => [
-                'created_at' => ['name' => 'created_at', 'dateFormat' => 'Y-m-d', 'meta' => ['date' => true]]
-            ],
-            'meta' => [
-                'date' => true
-            ]
-        ]);
+        $this->params['columns']['created_at'] = [
+            'name' => 'created_at',
+            'dateFormat' => 'Y-m-d',
+            'meta' => ['date' => true],
+        ];
+
+        $this->params['meta']['date'] = true;
 
-        $this->assertEquals($this->testModel->created_at->format('Y-m-d'), $response["data"][0]["created_at"]);
+        $response = $this->requestResponse();
+
+        $this->assertEquals(
+            $this->testModel->created_at->format('Y-m-d'),
+            $response->get('data')->first()->get('created_at')
+        );
     }
 
     /** @test */
     public function can_get_data_with_cent()
     {
-        $response = $this->requestResponse([
-            "columns" => [
-                'price' => ['name' => 'price', 'meta' => ['cents' => true]]
-            ],
-            'meta' => [
-                'cents' => true
-            ]
-        ]);
+        $this->params['columns']['price'] = ['name' => 'price', 'meta' => ['cents' => true]];
+
+        $this->params['meta']['cents'] = true;
```

Comment:
And what is the rule for a new line? When should we use newline?

Reply:
before & after ifs / closures, before return, and sometimes between long lines